File: C/estructuraCola.c
#include <iostream>
#include <queue>
using namespace std;

int main() {
    queue<int> cola;

    // Insertar elementos en la cola
    cola.push(1);
    cola.push(2);
    cola.push(3);

    // Mostrar el elemento frontal de la cola
    cout << "Elemento frontal de la cola: " << cola.front() << endl;

    // Eliminar el elemento frontal de la cola
    cola.pop();

    // Mostrar el elemento frontal de la cola después de la eliminación
    cout << "Elemento frontal de la cola después de la eliminación: " << cola.front() << endl;

    // Comprobar si la cola está vacía
    if (cola.empty()) {
        cout << "La cola está vacía." << endl;
    } else {
        cout << "La cola no está vacía." << endl;
    }

    return 0;
}

File: Php/estructuraCola.php
<?php
// Implementación de una cola (queue) en PHP usando un array
class Cola {
    private $items = [];

    public function estaVacia() {
        return empty($this->items);
    }

    public function encolar($item) {
        array_push($this->items, $item);
    }

    public function desencolar() {
        if (!$this->estaVacia()) {
            return array_shift($this->items);
        }
    }

    public function frente() {
        if (!$this->estaVacia()) {
            return $this->items[0];
        }
    }
}

// Crear una cola
$cola = new Cola();

// Insertar elementos en la cola
$cola->encolar(1);
$cola->encolar(2);
$cola->encolar(3);

// Mostrar el elemento frontal de la cola
echo "Elemento frontal de la cola: " . $cola->frente() . "\n";

// Eliminar el elemento frontal de la cola
$cola->desencolar();

// Mostrar el elemento frontal de la cola después de la eliminación
echo "Elemento frontal de la cola después de la eliminación: " . $cola->frente() . "\n";

// Comprobar si la cola está vacía
if ($cola->estaVacia()) {
    echo "La cola está vacía.\n";
} else {
    echo "La cola no está vacía.\n";
}
?>
